### PR TITLE
CR/CRD validation

### PR DESCRIPTION
**Pull request by jeremyary**
Per request, unique commits for:

1: only .gitignore IntelliJ add and dep/vendor changes
2: CR/CRD validation tests

### CODE REVIEW

**Comment by jeremyary** [comment]
Rather than complicate the discussion again with another CRD example, instead I've changed the struct-vs.-CRD validation test to include a list of omissions so that nested objects will no longer cause the test to fail. 
```
nestedOmissions := [...]string{
	// skip nested objects beneath each top-level object
	"/spec/storageDeviceSets/resources/",
	"/spec/storageDeviceSets/placement/nodeAffinity/",
	"/spec/storageDeviceSets/placement/podAffinity/",
	"/spec/storageDeviceSets/placement/podAntiAffinity/",
	"/spec/storageDeviceSets/placement/tolerations/",
	"/spec/storageDeviceSets/dataPVCTemplate/",
}
```
I'm still hoping this prompts a discussion on which of these nested objects (and how deeply recursive) the CRD should declare, but hopefully this better conveys what a 'more declarative' style would include.

File: test/crd_validation_test.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+func TestCompleteCRD(t *testing.T) {
+	root := "./../deploy/crds"
+	crdStructMap := map[string]interface{}{
+		"ocs_v1alpha1_ocsinitialization_crd.yaml": &v1alpha1.OCSInitialization{},
+		"ocs_v1alpha1_storagecluster_crd.yaml": &v1alpha1.StorageCluster{},
+	}
+	for crd, obj := range crdStructMap {
+		schema := getSchema(t, fmt.Sprintf("%s/%s", root, crd))
+		missingEntries := schema.GetMissingEntries(obj)
+		nestedOmissions := [...]string{
```
Comment (jarrpa) [inline test/crd_validation_test.go]: What would you say to defining this list as a const near the top of the file?

Comment (jeremyary) [reply test/crd_validation_test.go]: Updated. Additional paths from recent  CRD change merges also added.

**Comment by jeremyary** [comment]
> The go mod and new vendor changes should be in the other commit with the other vendor changes.

Bah sorry, forgot to rebase -i it first. Now I've taken a fresh copy of master and reapplied mod/tests just to make sure it jives well after so many merges. Should be cleaned up now.

**Pull request merged**